### ISSUE TITLE
[bitnami/mongodb] can't update helm deployment, because secret is being deleted before it is read

### ISSUE THREAD

**Issue by haroonb** [bitnami/charts#15173]
### Name and Version

bitnami/mongodb

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. `helm upgrade --install mongo --repo=https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami mongodb -n mongodb --wait --atomic --create-namespace -f mongodb-values.yaml --version=11.1.1`

```yaml
arbiter:
  enabled: true
architecture: replicaset
auth:
  enabled: true
  rootPassword: pass1234.
  rootUser: admin
global:
  namespaceOverride: mongodb
  storageClass: rook-ceph-block
image:
  tag: 4.4.13-debian-10-r19
metrics:
  enabled: true
  image:
    tag: 0.30.0-debian-10-r53
persistence:
  enabled: true
  size: 10Gi
  storageClass: rook-ceph-block
replicaCount: 3
replicaSetName: rs0
tls:
  enabled: false
useStatefulSet: false
```
3. try `helm upgrade --install mongo --repo=https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami mongodb -n mongodb --wait --atomic --create-namespace -f mongodb-values.yaml --version=11.1.1 --set auth.existingSecret="mongo-mongodb"`

add this to the previous values
```yaml
args:
  - "--wiredTigerCacheSizeGB=2"
```

```yaml
arbiter:
  enabled: true
args:
  - "--wiredTigerCacheSizeGB=2"
architecture: replicaset
auth:
  enabled: true
  rootPassword: pass1234.
  rootUser: admin
global:
  namespaceOverride: mongodb
  storageClass: rook-ceph-block
image:
  tag: 4.4.13-debian-10-r19
metrics:
  enabled: true
  image:
    tag: 0.30.0-debian-10-r53
persistence:
  enabled: true
  size: 10Gi
  storageClass: rook-ceph-block
replicaCount: 3
replicaSetName: rs0
tls:
  enabled: false
useStatefulSet: false
```
The container will be crashing giving this error message
```txt
Events:
  Type     Reason                  Age               From                     Message
  ----     ------                  ----              ----                     -------
  Normal   Scheduled               28s               default-scheduler        Successfully assigned mongodb/mongo-mongodb-2 to ops-w1
  Warning  FailedAttachVolume      28s               attachdetach-controller  Multi-Attach error for volume "pvc-9ecebda5-885f-407b-96db-aca4bd87d9e1" Volume is already exclusively attached to one node and can't be attached to another
  Normal   SuccessfulAttachVolume  21s               attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-9ecebda5-885f-407b-96db-aca4bd87d9e1"
  Normal   Pulled                  8s (x3 over 20s)  kubelet                  Container image "docker.io/bitnami/mongodb:4.4.13-debian-10-r19" already present on machine
  Warning  Failed                  8s (x3 over 20s)  kubelet                  Error: secret "mongo-mongodb" not found
  Normal   Pulled                  8s (x3 over 20s)  kubelet                  Container image "docker.io/bitnami/mongodb-exporter:0.30.0-debian-10-r53" already present on machine
  Warning  Failed                  8s (x3 over 20s)  kubelet                  Error: secret "mongo-mongodb" not found
```

### What is the expected behavior?

The upgrade process to read the credentials from the secret and upgrade the cluster, without deleting the secret.

### What do you see instead?

The process removed the secret, before it could read it, to get the needed secret/credentials.

### Additional information

The secret `mongo-mongodb` was created automatically by the bitnami chart.
I tried to update it without passing the secret, but I got the error.
```
helm upgrade --install mongo --repo=https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami mongodb -n mongodb --reset-values --wait --atomic --create-namespace -f mongodb-values.yaml --version=11.1.1
E0228 09:30:23.740224   50099 memcache.go:255] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0228 09:30:23.750669   50099 memcache.go:106] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
Error: UPGRADE FAILED: execution error at (mongodb/templates/NOTES.txt:204:4):
PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
                 Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
                 Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases

    'auth.replicaSetKey' must not be empty, please add '--set auth.replicaSetKey=$MONGODB_REPLICA_SET_KEY' to the command. To get the current value:

        export MONGODB_REPLICA_SET_KEY=$(kubectl get secret --namespace "mongodb" mongo-mongodb -o jsonpath="{.data.mongodb-replica-set-key}" | base64 --decode)
```
I read up in the link, that the letting helm create a random secret is `not recommended for production environments`, however it's very convenient and I would still assume, that the upgrade process would not delete the previously generated secret.

Exporting the secret to the bash environment is not really possible, because the main deployment method for our production environment is through playbooks and the helm module doesn't seem to support this function.

Can you please clarify, if this is intended or a bug.

Thank you.

**Comment by javsalgar** [comment]
Hi,

Could you specify which helm version you are using?

**Comment by haroonb** [comment]
Hello @javsalgar ,

**Version of Helm and Kubernetes**:
```
helm version
version.BuildInfo{Version:"v3.11.0", GitCommit:"472c5736ab01133de504a826bd9ee12cbe4e7904", GitTreeState:"clean", GoVersion:"go1.18.10"}
```
```
kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.1", GitCommit:"3ddd0f45aa91e2f30c70734b175631bec5b5825a", GitTreeState:"clean", BuildDate:"2022-05-24T12:26:19Z", GoVersion:"go1.18.2", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.10", GitCommit:"5c1d2d4295f9b4eb12bfbf6429fdf989f2ca8a02", GitTreeState:"clean", BuildDate:"2023-01-18T19:08:10Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
```

Thank you.

**Comment by aoterolorenzo** [comment]
Hi haroonb,

Seems there could be a bug in the changes introduced in 

#15093 and #15311 seems also related to this.

The team is investigating it.

**Comment by aoterolorenzo** [comment]
@haroonb the fix is now merged. Could you try it? Your version is fixed to 11.1.1 so it's impossible to say.

**Comment by aoterolorenzo** [comment]
I will open a task for the team since it seems to need a deeper look. We will reach you back with the outcome as soon as our workload allow as to address it
